src/Magnum/Platform/GlfwApplication.cpp

```diff
@@ -0,0 +1,213 @@
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "GlfwApplication.h"
+
+#include <tuple>
+
+#include "Magnum/Version.h"
+#include "Magnum/Platform/Context.h"
+#include "Magnum/Platform/ScreenedApplication.hpp"
+
+namespace Magnum { namespace Platform {
+
+typedef GlfwApplication::Configuration::Flag Flag;
+typedef GlfwApplication::Configuration::Flags Flags;
+typedef GlfwApplication::Configuration::WindowFlag WindowFlag;
+typedef GlfwApplication::Configuration::WindowFlags WindowFlags;
+
+GlfwApplication* GlfwApplication::_instance = nullptr;
+
+#ifndef DOXYGEN_GENERATING_OUTPUT
+GlfwApplication::GlfwApplication(const Arguments& arguments): GlfwApplication{arguments, Configuration{}} {}
+#endif
+
+GlfwApplication::GlfwApplication(const Arguments& arguments, const Configuration& configuration): GlfwApplication{arguments, nullptr} {
+    createContext(configuration);
+}
+
+GlfwApplication::GlfwApplication(const Arguments& arguments, std::nullptr_t)
+    : _context{new Context{NoCreate, arguments.argc, arguments.argv}},
+      _needsRedraw(true)
+{
+    /* Save global instance */
+    _instance = this;
+
+    /* Init GLFW */
+    glfwSetErrorCallback(staticErrorCallback);
+
+    if(!glfwInit()) {
+        Error() << "Could not initialize GLFW";
+        std::exit(8);
+    }
+}
+
+void GlfwApplication::createContext() { createContext({}); }
+
+void GlfwApplication::createContext(const Configuration& configuration) {
+    if(!tryCreateContext(configuration)) std::exit(1);
+}
+
+bool GlfwApplication::tryCreateContext(const Configuration& configuration) {
+    CORRADE_ASSERT(_context->version() == Version::None, "Platform::GlfwApplication::tryCreateContext(): context already created", false);
+
+    const WindowFlags NONE = WindowFlags{};
+
+    /* Window flags */
+    GLFWmonitor* monitor = nullptr; /* Needed for setting fullscreen */
+    if ((configuration.windowFlags() & WindowFlag::Fullscreen) != NONE) {
+        monitor = glfwGetPrimaryMonitor();
+        glfwWindowHint(GLFW_AUTO_ICONIFY, ((configuration.windowFlags() & WindowFlag::AutoIconify) != NONE) ? GLFW_TRUE : GLFW_FALSE);
+    } else {
+        const Configuration::WindowFlags& flags = configuration.windowFlags();
+        glfwWindowHint(GLFW_RESIZABLE, ((flags & WindowFlag::Resizeable) != NONE) ? GLFW_TRUE : GLFW_FALSE);
+        glfwWindowHint(GLFW_VISIBLE, ((flags & WindowFlag::Hidden) != NONE) ? GLFW_FALSE : GLFW_TRUE);
+        glfwWindowHint(GLFW_MAXIMIZED, ((flags & WindowFlag::Maximized) != NONE) ? GLFW_TRUE : GLFW_FALSE);
+        glfwWindowHint(GLFW_ICONIFIED, ((flags & WindowFlag::Minimized) != NONE) ? GLFW_TRUE : GLFW_FALSE);
+        glfwWindowHint(GLFW_FLOATING, ((flags & WindowFlag::Floating) != NONE) ? GLFW_TRUE : GLFW_FALSE);
```

Comment:
*Uh*, thank god :D I was very unhappy with that code.

```
glfwWindowHint(GLFW_RESIZABLE, flags >= WindowFlag::Resizeable);
```

So beautiful! :dizzy_face: